docs/testnet-guardian-runbook.md

```diff
@@ -0,0 +1,130 @@
+
+## QA-net
+
+The role of a testnet guardian for a release begins after the code-freeze. The following is the list of tasks that a testnet guardian would be responsible for:
+
+### Release branch
+
+Cut a release branch after the code-freeze(The naming convention for beta-testnets is `release/0.0.<release_number>-beta`. For example, `release/0.0.13-beta`)
+
+### Build
+
+The latest commit on the release branch should have successful builds for both macOS and Linux.
+Go to the CI workflow for the specific commit and look for `build- macos` and `build-artifacts--testnet_postake_medium_curves` jobs for mac and linux builds respectively.
+
+##### Update brew package
+To update brew package,
+
+1. Download the homebrew binary package from gcloud at the location specified in the `Copy artifacts to cloud` task in the `build-macOS` CI job. For example `gs://network-debug/382998/build/homebrew-coda.tar.gz`
+2. Upload it to our public s3 bucket at `https://s3-us-west-2.amazonaws.com/packages.o1test.net`
+3. In `coda.rb` in `CodaProtocol/homebrew-coda`,
+        1. update the url with the location of the new tarball
+        2. update the hash. The file containing the hash is uploaded to gcloud as well, for example `gs://network-debug/382998/build/homebrew-coda.tar.gz.sha256`
+        3. increment the revision number
+
+##### Update apt package
+ 1. Create an annotated tag for the latest commit on develop after code freeze. For example: `git tag -m "release 0.0.13" 0.0.13`
+ 2. push the tag `git push origin 0.0.13`
+ 3. Push the changes from develop on to master
+
+##### Publish the location of the binaries
+In addition to updating the brew and apt packages, publish the locations to the binaries on the discord channel `#QA-net details` channel for the rest of the team
+1. macOS binary
+2. linux binary
+3. docker image
+4. Name of the release branch
+
+### Deploy
+
+Deploy a QA-net (currently a 200-node network) with the new binaries. Ensure gossip logs are enabled by setting the daemon flags `log-snark-work-gossip`, `log-txn-pool-gossip`, and `log-received-blocks`.
+
+Instructions for configuring and deploying a QA-net is roughly documented here: https://www.notion.so/codaprotocol/Testnet-deployment-notes-c1fbf8b8be9343b480f644c22d61ad85
+
+### Test
+
+After the QA-net is deployed, perform the following sanity-check:
+
+##### Sanity check
+
+1. Search for any crashes by querying the logs on StackDriver. The nodes are configured to restart upon crash for some of the known issues (#4295). Create issues for new ones.
+
+2. Run `scripts/ezcheck.sh` to check if all the nodes are synced and have the same state hash at their best tips.
+
+3. Query the logs to confirm that the nodes are creating and receiving blocks, transaction-pool diffs, and snark-pool diffs.
```

Comment:
I usually check that block producers can see txns and proofs when they build new staged ledger diffs. I typically only check the receiving logs if something is not correct at that layer. I do it like this because sometimes the traffic is received, but isn't "valid" against the best tip for one reason or another.